utilities/merge_operators/sortlist.cc

```diff
@@ -0,0 +1,115 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#include <exception>
+#include <iostream>
+#include <memory>
+
+#include "rocksdb/merge_operator.h"
+#include "rocksdb/slice.h"
+#include "utilities/merge_operators.h"
+#include "utilities/merge_operators/sortlist.h"
+
+using rocksdb::Logger;
+using rocksdb::MergeOperator;
+using rocksdb::Slice;
+
+namespace rocksdb {
+
+// Merge operator that picks the maximum operand, Comparison is based on
+// Slice::compare
+bool SortList::FullMergeV2(const MergeOperationInput& merge_in,
+                           MergeOperationOutput* merge_out) const {
+  (void)merge_out;
+  std::vector<int> left;
+  for (Slice slice : merge_in.operand_list) {
+    std::vector<int> right;
+    make_vector(right, slice);
+    left = merge(left, right);
+  }
+  for (int i = 0; i < (int)left.size() - 1; i++) {
+	merge_out->new_value.append(std::to_string(left[i])).append(",");
+  }
+  merge_out->new_value.append(std::to_string(left.back()));
+  return true;
+}
+
+bool SortList::PartialMerge(const Slice& /*key*/, const Slice& left_operand,
+                            const Slice& right_operand, std::string* new_value,
+                            Logger* /*logger*/) const {
+  std::vector<int> left;
+  std::vector<int> right;
+  make_vector(left, left_operand);
+  make_vector(right, right_operand);
+  left = merge(left, right);
+  for (int i = 0; i < (int)left.size() - 1; i++) {
+    new_value->append(std::to_string(left[i])).append(",");
+  }
+  new_value->append(std::to_string(left.back()));
+  return true;
+}
+
+bool SortList::PartialMergeMulti(const Slice& /*key*/,
+                                 const std::deque<Slice>& operand_list,
+                                 std::string* new_value,
+                                 Logger* /*logger*/) const {
+  (void)operand_list;
+  (void)new_value;
+  return true;
+}
+
+const char* SortList::Name() const { return "MergeSortOperator"; }
+
+void SortList::make_vector(std::vector<int>& operand, Slice slice) const {
+  int errors = 0;
+  do {
+    const char* begin = slice.data_;
+    while (*slice.data_ != ',' && *slice.data_) slice.data_++;
+    try {
+      operand.push_back(std::stoi(std::string(begin, slice.data_)));
+    } catch (...) {
+      //			std::cout << "Malformed string: " <<
```

Comment:
Had to add this because sometimes the data is garbage, not sure why. I also see compilation failed in one of the jobs with error: utilities/merge_operators/sortlist.cc:72:14: error: exception handling disabled, use -fexceptions to enable. Any suggestions?

Reply:
The bug is fixed and does this file still need to be moved to a different location?